src/Models/BaseElement.php

```diff
@@ -679,6 +679,38 @@ public function getAnchor()
         return $this->anchor = $result;
     }
 
+    /**
+     * Get anchors in this block's content.
+     * Used to populate the "anchor on a page" link in the WYSIWYG
+     *
+     * By default, this finds anchors in any HTMLText field on the block, but
+     * this method should be overridden if anchors are provided in other ways
+     * for this block or if not all HTMLText fields for this block are
+     * displayed on the front-end.
+     */
+    public function getAnchorsInContent(): array
+    {
+        $anchors = [$this->getAnchor()];
+        $anchorRegex = "/\\s+(name|id)\\s*=\\s*([\"'])([^\\2\\s>]*?)\\2|\\s+(name|id)\\s*=\\s*([^\"']+)[\\s +>]/im";
+        $allFields = DataObject::getSchema()->fieldSpecs($this);
+        foreach ($allFields as $field => $fieldSpec) {
+            $fieldObj = $this->owner->dbObject($field);
+            if ($fieldObj instanceof DBHTMLText) {
+                $parseSuccess = preg_match_all($anchorRegex, $fieldObj->getValue() ?? '', $matches);
+                if ($parseSuccess >= 1) {
+                    $fieldAnchors = array_values(array_filter(
+                        array_merge($matches[3], $matches[5])
```

Comment:
I would suggest to add name for the group in regular expression, like `\s+(name|id)\s*?=\s*?(\s*?[\"']\s*?)(?<token>\s*?[^\2\s>]*?\s*?)\2` and replace $matches[3] to $match['token']. Also, remove $matches[5], because the 5th group returns only quotes.

Reply:
The fifth group actually returns anything _other than_ quotes. `([^"']+)` The `^` there turns it into "anything other than".

I don't think adding named groups would be beneficial here - both of the two groups we are interested in just contain anchor names, so the two groups would have the same name which for obvious reasons won't work.

Reply:
`([^"']+)` -  Sorry, my fault.
But anyway, it looks like it should find string something like this `<p name=my_p_link_2>My paragraph</p><p`, but not id / name title

Reply:
Yes, exactly. So in that example `my_p_link_2` would be the anchor.

Reply:
I'll test second part of this condition and if it will be needed I'll create new ticket for enchancement.